### PR TITLE
Clippy pass

### PR DESCRIPTION
**Pull request by pbougue**
Recommend reading all but bragi_test.
Then bragi_*_test alone:
* add a handler for bragi
* separate tests in fn
* outsource utils fn
* split in different files

TODO:
- [x] answer 2 questions on comment left in bragi_test (last commit)

### CODE REVIEW

File: tests/bragi_test.rs [hunk 1]
```diff
@@ -455,32 +368,19 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
     // for exemple there are parkings without name (but with the tag "anemity" = "Parking"),
     // we should be able to query them
-    let geocodings = get_results(bragi_get("/autocomplete?q=Parking"));
+    let geocodings = get_results(bragi.get("/autocomplete?q=Parking"));
     let types = get_types(&geocodings);
     assert_eq!(count_types(&types, "poi"), 5);
 
-    // we search for a POI (id = 2561223) with a label but an empty ?, it should be filtered)
-    let geocodings = get_results(bragi_get("/autocomplete?q=ENSE3 site Ampère"));
+    // we search for a POI (id = 2561223) with a label but an empty <?????>, it should be filtered)
```
Comment (pbougue) [inline tests/bragi_test.rs]: Don't know what's empty ( @azime maybe you do)

File: tests/bragi_test.rs [hunk 2]
```diff
@@ -510,19 +413,171 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
+                        vec!["--input=./tests/fixtures/osm_fixture.osm.pbf".into(),
+                             "--import-way".into(),
+                             "--level=8".into(),
+                             format!("--connection-string={}", es_wrapper.host())],
+                        &es_wrapper);
+
+    zip_code_test(&bragi);
+    zip_code_street_test(&bragi);
+    zip_code_admin_test(&bragi);
+    // zip_code on addr
```
Comment (pbougue) [inline tests/bragi_test.rs]: Is there a missing test here? (or we should remove that comment)

File: src/boundaries.rs [hunk 3]
```diff
@@ -139,9 +139,9 @@ pub fn build_boundary(relation: &osmpbfreader::Relation,
             obj
         })
         .filter_map(|way_obj| way_obj.way())
-        .map(|way| get_nodes(&way, objects))
+        .map(|way| get_nodes(way, objects))
         .filter(|nodes| nodes.len() > 1)
-        .map(|nodes| BoundaryPart::new(nodes))
+        .map(BoundaryPart::new)
```
Comment (antoine-de) [inline src/boundaries.rs]: even if it's shoter, I think it's a bit more difficult to read

File: tests/bragi_test.rs [hunk 4]
```diff
@@ -510,19 +413,171 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
+
+    // *********************************
+    // We load bano files
+    // *********************************
+    ::launch_and_assert(bano2mimir,
+                        vec!["--input=./tests/fixtures/sample-bano.csv".into(),
+                             format!("--connection-string={}", es_wrapper.host())],
+                        &es_wrapper);
+
+    status_test(&bragi);
```
Comment (antoine-de) [inline tests/bragi_test.rs]: hum if we split this into several functions I think we need to do some stuff to make the data loaded clearer.

I would have either split this into several integration tests (ie a new file)
or split this into several function but with an ES cleanup between each function (and so one function by 'datasets' and not by tests)

File: tests/bragi_test.rs [hunk 5]
```diff
@@ -510,19 +413,171 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
+    app: rustless::Application,
+}
+
+impl BragiHandler {
+    pub fn new(url: String) -> BragiHandler {
+        let api = bragi::api::ApiEndPoint { es_cnx_string: url }.root();
+        BragiHandler { app: rustless::Application::new(api) }
+    }
+
+    pub fn params_validation(&self, q: &str) -> iron::IronResult<iron::Response> {
```
Comment (antoine-de) [inline tests/bragi_test.rs]: isn't it a bit strange as the function name ?

File: tests/bragi_test.rs [hunk 6]
```diff
@@ -510,19 +413,171 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
+        iron_test::request::get(&format!("http://localhost:3000{}", q),
+                                iron::Headers::new(),
+                                &self.app)
+    }
+
+    pub fn get(&self, q: &str) -> iron::Response {
+        self.params_validation(q).unwrap()
```
Comment (antoine-de) [inline tests/bragi_test.rs]: wouldn't it be better for this method to return a `get_results(self.params_validation(q).unwrap())`

I think it's the most common use case

File: src/boundaries.rs (hunk 3 above)
Comment (TeXitoi) [inline src/boundaries.rs]: I don't agree

File: libs/bragi/src/lib.rs [hunk 7]
```diff
@@ -55,7 +55,6 @@ extern crate rs_es;
 use iron::Iron;
 use rustless::Application;
 
-#[macro_use]
 extern crate mdo;
```
Comment (TeXitoi) [inline libs/bragi/src/lib.rs]: mdo without the macro is quite useless, I think you can remove the export, it should not be used.

File: tests/bano2mimir_test.rs [hunk 8]
```diff
@@ -67,7 +67,7 @@ pub fn bano2mimir_sample_test(es_wrapper: ::ElasticSearchWrapper) {
          s =<< s.as_object();
          ret ret(s.keys().cloned().collect())
      }
-        .unwrap_or(vec![]);
+        .unwrap_or_else(|| vec![]);
```
Comment (TeXitoi) [inline tests/bano2mimir_test.rs]:
```rust
.unwrap_or_else(Vec::new);
```
?

File: tests/bragi_test.rs [hunk 9]
```diff
@@ -217,13 +142,14 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
 
-    // zip_code and name of street
-    let all_20 = get_results(bragi_get("/autocomplete?q=77000 Lotissement le Clos de Givry"));
+fn zip_code_street_test(bragi: &BragiHandler) {
+    let all_20 = get_results(bragi.get("/autocomplete?q=77000 Lotissement le Clos de Givry"));
     assert_eq!(all_20.len(), 1);
     assert!(get_values(&all_20, "postcode").iter().all(|r| *r == "77000"));
 
-    let boundary = all_20[0].get("administrative_regions").unwrap().pointer("/0/boundary").unwrap();
+    let boundary = all_20[0]["administrative_regions"].pointer("/0/boundary").unwrap();
```
Comment (TeXitoi) [inline tests/bragi_test.rs]:
```rust
let boundary = all_20.pointer("/0/administrative_regions/0/boundary").unwrap();
```
maybe?

Comment (TeXitoi) [inline tests/bragi_test.rs]: That's a `Vec<BTreeMap<String, Value>>`, forget this comment.

File: src/boundaries.rs (hunk 3 above)
Comment (pbougue) [inline src/boundaries.rs]: the "no argument" version wins by 3-2, I let it as is

**Comment by pbougue** [comment]
warnings left in nightly on travis, will have a look at that

**Pull request merged**